### PR TITLE
Port to Python3 and GTK 3+ Bug Fixes 

### PR DESCRIPTION
**Pull request by srevinsaju**
wrt GCI 2019
Ran pytests and confirmed.

### CODE REVIEW

File: page.py [hunk 1]
```diff
@@ -12,7 +12,8 @@
 # Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 # Boston, MA 02111-1307, USA.
 
-
+import gi
+gi.require_version('Gtk', '3.0')
```
Comment (chimosky) [inline page.py]: You've already added a `gi.require_version` in `LetterMatch.py` and you need just one.

**Comment by srevinsaju** [comment]
Test

File: page.py (hunk 1 above)
Comment (srevinsaju) [reply page.py]: Oh ok, I will remove that

**Comment by chimosky** [comment]
Tested, logs contain.

```
/home/ibiam/Activities/lettermatch/page.py:253: Warning: Source ID 1043 was not found when attempting to remove it
  GObject.source_remove(self.timeout)
```

**Comment by chimosky** [comment]
@srevinsaju your commit message in 493a4b3 should address the issue you fixed.

**Comment by srevinsaju** [comment]
@chimosky Is it working or does it crash?
I will change the commit message

**Comment by srevinsaju** [comment]
Yay changed the message
Thanks @chimosky

**Comment by srevinsaju** [comment]
With 62ff9ce05532999f9fa404f5bae923e1ed2064f1 the above mentioned
> /home/ibiam/Activities/lettermatch/page.py:253: Warning: Source ID 1043 was not found when attempting to remove it
  GObject.source_remove(self.timeout)

warning is removed, pure code ready!

**Comment by Aniket21mathur** [comment]
I am getting this traceback
```
File "/home/aniket/activities/lettermatch/page.py", line 438, in svg_str_to_pixbuf
    pl.write(svg_string)
TypeError: Item 0: Must be number, not str
```

**Comment by quozl** [comment]
Thanks.  I've cherry-picked d9a3a39 and 5eb6050 as part of 938fd6c3de1eee3a39ad57f8bffb9d51fc1bfadf.

**Pull request closed without merging**